internal/appsec/limiter_test.go

```diff
@@ -0,0 +1,305 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+//go:build appsec
+// +build appsec
+
+package appsec
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLimiterUnit(t *testing.T) {
+	startTime := time.Now()
+
+	t.Run("no-ticks-1", func(t *testing.T) {
+		l := NewTestTicker(1, 100)
+		l.start(startTime)
+		defer l.stop()
+		// No ticks between the requests
+		require.True(t, l.Allow(), "First call to limiter.Allow() should return True")
+		require.False(t, l.Allow(), "Second call to limiter.Allow() should return False")
+	})
+
+	t.Run("no-ticks-2", func(t *testing.T) {
+		l := NewTestTicker(100, 100)
+		l.start(startTime)
+		defer l.stop()
+		// No ticks between the requests
+		for i := 0; i < 10; i++ {
+			require.True(t, l.Allow(), "Call to limiter.Allow() should return True")
+		}
```

Comment:
Crossing the limit is worth it again imo:
```suggestion
		for i := 0; i < 100; i++ {
			require.True(t, l.Allow())
		}
		require.False(t, l.Allow())
```